drivers/i2s/i2s_nrfx.c

```diff
@@ -0,0 +1,788 @@
+		LOG_ERR("Inappropriate trigger (%d/%d), active stream(s): %d",
+			cmd, dir, drv_data->active_dir);
+		return -EINVAL;
+	}
+
+	if (!cmd_allowed) {
+		LOG_ERR("Trigger %d not allowed in state %d",
+			cmd, drv_data->state);
+		return -EIO;
+	}
+
+	switch (cmd) {
+	case I2S_TRIGGER_START:
+		drv_data->stop = false;
+		drv_data->active_dir = dir;
+		drv_data->last_tx_buffer = NULL;
+		{
+			int ret = start_transfer(dev, dir);
+
+			if (ret < 0) {
+				return ret;
+			}
+		}
+		drv_data->state = I2S_STATE_RUNNING;
+		break;
+
+	case I2S_TRIGGER_STOP:
+		drv_data->state = I2S_STATE_STOPPING;
+		drv_data->stop = true;
+		break;
+
+	case I2S_TRIGGER_DRAIN:
+		drv_data->state = I2S_STATE_STOPPING;
+		/* If only RX is active, DRAIN is equivalent to STOP. */
+		drv_data->stop = (drv_data->active_dir == I2S_DIR_RX);
+		break;
+
+	case I2S_TRIGGER_DROP:
+		nrfx_i2s_stop();
+		purge_queue(drv_data, dir);
+		drv_data->state = I2S_STATE_READY;
+		break;
+
+	case I2S_TRIGGER_PREPARE:
+		purge_queue(drv_data, dir);
+		drv_data->state = I2S_STATE_READY;
+		break;
+
+	default:
+		LOG_ERR("Invalid trigger: %d", cmd);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static const struct i2s_driver_api i2s_nrf_drv_api = {
+	.configure = i2s_nrfx_configure,
+	.config_get = i2s_nrfx_config_get,
+	.read = i2s_nrfx_read,
+	.write = i2s_nrfx_write,
+	.trigger = i2s_nrfx_trigger,
+};
+
+#define I2S(idx) DT_NODELABEL(i2s##idx)
+
+#define I2S_PIN(idx, name) NRF_DT_PSEL(I2S(idx), name##_pin, name##_gpios, \
+				       NRFX_I2S_PIN_NOT_USED)
+#define I2S_STREAM(idx, dir, sem_init, sem_max)				\
+	{								\
+		.queue = {						\
+			.items = dir##_queue##idx,			\
+			.size  = ARRAY_SIZE(dir##_queue##idx),		\
+		},							\
+		.sem = Z_SEM_INITIALIZER(i2s_nrfx_data##idx.dir.sem,	\
+					sem_init, sem_max),		\
+	}
+#define I2S_NRFX_INST(idx)						     \
+	NRF_DT_PSEL_CHECK_EXACTLY_ONE(I2S(idx),				     \
+				      sck_pin, "sck-pin",		     \
+				      sck_gpios, "sck-gpios");		     \
+	NRF_DT_PSEL_CHECK_EXACTLY_ONE(I2S(idx),				     \
+				      lrck_pin, "lrck-pin",		     \
+				      lrck_gpios, "lrck-gpios");	     \
+	NRF_DT_PSEL_CHECK_NOT_BOTH(I2S(idx),				     \
+				   mck_pin, "mck-pin",			     \
+				   mck_gpios, "mck-gpios");		     \
+	NRF_DT_PSEL_CHECK_NOT_BOTH(I2S(idx),				     \
+				   sdout_pin, "sdout-pin",		     \
+				   sdout_gpios, "sdout-gpios");		     \
+	NRF_DT_PSEL_CHECK_NOT_BOTH(I2S(idx),				     \
+				   sdin_pin, "sdin-pin",		     \
+				   sdin_gpios, "sdin-gpios");		     \
+	static int i2s_nrfx_init##idx(const struct device *dev)		     \
+	{								     \
+		IRQ_CONNECT(DT_IRQN(I2S(idx)), DT_IRQ(I2S(idx), priority),   \
+			    nrfx_isr, nrfx_i2s_irq_handler, 0);		     \
+		irq_enable(DT_IRQN(I2S(idx)));				     \
+		return 0;						     \
+	}								     \
+	static void data_handler##idx(nrfx_i2s_buffers_t const *p_released,  \
+				      uint32_t status)			     \
+	{								     \
+		data_handler(DEVICE_DT_INST_GET(idx), p_released, status);   \
```

Comment:
I think you mean `DEVICE_DT_GET(I2S(idx))`, right?

Reply:
Yes. Fixed.